### PR TITLE
Add contrib/json.arrai for working with JSON data

### PR DESCRIPTION
**Pull request by orlade-anz**
The standard, canonical JSON structure is a pain to work with. In most cases the decoded data need not be reversible, and follows a schema that makes a tuple more appropriate than a map. This PR adds functions to apply these simplifications.

### CODE REVIEW

File: contrib/json.arrai [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+# Returns a "simplified" structure, replacing decoded tuples with their internal values.
+# Intended for cases in which the types of null values is not interesting.
+let rec simplify = \j
+    cond j {
+        [...]: j >> simplify(.),
+        (:a): simplify(a),
+        (:s): simplify(s),
+        (): {},
+        _: cond {
+            j < 99999999999999999999: j,
```
Comment (nofun97) [inline contrib/json.arrai:12]: is this meant to check if `j` is a number?

File: contrib/json.arrai [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# Helper functions for working with JSON data.
+
+# Returns a "simplified" structure, replacing decoded tuples with their internal values.
+# Intended for cases in which the types of null values is not interesting.
+let rec simplify = \j
```
Comment (nofun97) [inline contrib/json.arrai:5]: also handle `(:b)` for boolean values.

**Review by nofun97 — CHANGES_REQUESTED** [review]

File: contrib/json.arrai (hunk 1 above)
Comment (orlade-anz) [reply contrib/json.arrai:12]: Yeah. I forget now what the problem was, but I couldn't figure out how else to do it.

Comment (orlade-anz) [reply contrib/json.arrai:15]: Added a comment as well.

**Pull request merged**